kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/client/grpc/GrpcStreamSender.java

```diff
@@ -30,16 +32,34 @@ class GrpcStreamSender<M, P> implements StreamSender<M> {
 
   @Override
   public void sendNext(M msg) {
-    grpcObserver.onNext(protoFactory.apply(msg));
+    try {
+      grpcObserver.onNext(protoFactory.apply(msg));
+    } catch (Exception e) {
+      grpcObserver.onError(e);
```

Comment:
related to my q below on how errors are surfaced, this `grpcObserver` is the grpc side of the client->server stream. Does that propagate the error to our side of the server->client stream?